lr-mgba.sh: enable lr-mgba for GB and GBC (SGB also supported)

hhromic opened the pull request:
As mentioned in #2295, `lr-mgba` is mature enough now to support GB/GBC/SGB as well as GBA.
I've been using it for several weeks now and all works fine. The emulator detects the system type automatically and can use all BIOSes available, even Super Gameboy.
The emulation speed in my standard-configured RPI3 is very satisfactory too.
This is for the `master` upstream repository of `lr-mgba`, used when you install it from source.
I strongly suggest the binary version in RetroPie is updated too to make all work out-of-the-box.

The following BIOS files were used for testing each system type:

    a860e8c0b6d573d191e4ec7db1b1e4f6  gba_bios.bin
    32fbbd84168d3482956eb3c5051637f5  gb_bios.bin
    dbfce9db9deaa2567f6a84fde55f9680  gbc_bios.bin
    d574d4f9c12f305074798f54c091a8b4  sgb_bios.bin

I kept the original logic used to set the emulator as default, e.g. set it as default in non `armv6` platforms.
This should keep users of slow hardware unaffected if they are using `lr-gambatte` already.

hhromic commented:
If this change gets merged and the binary updated, I will update the Wiki to include BIOS file information too.

psyke83 commented on scriptmodules/libretrocores/lr-mgba.sh:
```diff
@@ -40,11 +40,12 @@ function install_lr-mgba() {
     isPlatform "armv6" && def=0
-    addEmulator $def "$md_id" "gba" "$md_inst/mgba_libretro.so"
-    addSystem "gba"
+    for system in gb gbc gba; do
```
You should ensure the ```system``` variable is local scope

hhromic replied:
oh forgot that, thanks for spotting.

psyke83 commented on scriptmodules/libretrocores/lr-mgba.sh:
```diff
@@ -40,11 +40,12 @@ function install_lr-mgba() {
+    for system in gb gbc gba; do
+        mkRomDir "$system"
+        ensureSystemretroconfig "$system"
+        addEmulator "$def" "$md_id" "$system" "$md_inst/mgba_libretro.so"
```
I'm not certain if @joolswills wants this patch included at all, but if so, perhaps he can chime in on whether the previous defaults should be preserved, or if all three systems should use ```lr-mgba``` as set by this patch (I think not).

hhromic commented:
Perhaps the correct thing to do is to follow the same approach/logic than for `lr-genesis-plus-gx` and `lr-picodrive`, two competing emulators for various Sega 8/16 bits consoles:

`lr-picodrive` is used as default for: **megadrive mastersystem segacd sega32x**.
`lr-genesis-plus-gx` is used as default for: **gamegear sg-1000**.
`lr-genesis-plus-gx` is used as alternative for:  **megadrive mastersystem segacd**.

Then I can rework the patch to achieve this style of behaviour for the Game Boy consoles:

`lr-gambatte` is used as default for: **gb gbc**
`lr-mgba` is used as default (in non `armv6`) for: **gba**
`lr-mgba` is used as alternative for: **gb gbc**
`lr-gpsp` is used as default (in `armv6`) for: **gba**
`lr-vba-next` is used as alternative for: **gba**

How does this sound?

hhromic commented:
Hi @psyke83 , @joolswills , did you have any chance to review this PR?
Let me know of your thoughts to keep working on it and get it merged.
If you are not happy with the defaults handling I propose (check the edited comment), we can discard that, but at least I think we should allow lr-mgba to be an alternative for GB/GBC even if it is not their default emulator.
Cheers and thanks for all your work!

joolswills commented:
I will add this as an alternative but I am not going to change any defaults currently.

hhromic commented:
Good stuff. Do you want me to refactor this PR accordingly?

joolswills commented:
If you would like - please squash the commits also. alternatively I am happy to implement it when I next have some time.

hhromic commented:
No problem from my side, I was also planning to squash the commits. Will update shortly. Thanks for the feedback.

hhromic commented:
Update the PR now:
* Updated the module description and help.
* Added GB/GBC as alternative but not default.
* Kept (as originally) GBA as default only in non armv6 platforms.
Thanks!

joolswills commented:
Thanks

hhromic commented:
@joolswills , before I forget , it is possible to also update the binary for lr-mgba to be up-to-date to its current master? Thanks!

joolswills commented:
Can do. It was updated a few weeks ago btw.

hhromic commented:
Great, should be fine then. When you have time, it won't hurt if you update again just in case.
Thanks a million.

hhromic commented:
Hi @joolswills , sorry to bring this up again. Looks like users in the forum are needing to update `lr-mgba` from source to get the latest features enabled recently: https://retropie.org.uk/forum/post/146141
Maybe a refresh of the binary is a good idea? Cheers!
